tests/s3_tester.c

```diff
@@ -785,6 +787,13 @@ int aws_s3_tester_send_meta_request(
 
     ASSERT_TRUE(meta_request != NULL);
 
+    if (flags & AWS_S3_TESTER_SEND_META_REQUEST_CANCEL) {
```

Comment:
Could we do something more deterministic here?  For example, maybe a cancel based on the aws_s3_auto_ranged_put_state?

Reply:
Yeah, I don't like the testing here, but probably will just add TODOs for the proper testing.